[WIP] add owntone (aka forked-daapd)

hgy59 opened the pull request:
_Motivation:_  owntone (former forked-daadp) is a rewrite of  mt-daapd (Firefly Media Server) and requested a long time ago.
_Linked issues:_  closes #1160

### Checklist
- [x] Build rule `all-supported` completed successfully
- [ ] New installation of package completed successfully
- [ ] ~Replace cross/ffmpeg dependency by cross/libav (daapd needs audio only)~
- [ ] Add service setup scripts
- [x] solve [libssp.so for avahi](https://github.com/SynoCommunity/spksrc/issues/1160#issuecomment-396066216)

### Remarks
- add forked-daadp v27.4
- enable unlock notify in cross/sqlite

This is really a huge package with a lot of dependencies. It will take some time to finalize the package integration.

hgy59 committed:
add forked-daadp

- add forked-daadp v27.2
- add dependencies
- enable unlock notify in cross/sqlite

taku0220 commented:
Hi,@hgy59.
Thanks to this PR, I was able to successfully build forked-daapd v27.4.
I'm still in the process of testing it, but it seems to be working fine.
My test repository is [here](https://github.com/taku0220/spksrc/tree/forked-daapd_27.4_rebase).
You will need to install libgcrypt20-dev, libgpg-error-dev and antlr3 in your development environment.
Is there anything I can do to help?

NAS Model: DS220j
NAS Architecture: rtd1296
DSM version: DSM 6.2.4-25556

It seems that the name of forked-daapd will be changed to owntone-server in the next release version.

I am not very good at English, so I apologize in advance.

hgy59 committed:
rename package to owntone

- source is still called forked-daapd but repository is renamed to owntone

hgy59 committed:
update owntone to v28.1

- reintroduce native/glib for cross/avahi
- fix ffmpeg on demand dependencies
- forked-daapd is fully renamed to owntone

hgy59 committed:
add initial service setup

- add default config adjusted for DSM
- add libssp from toolchains

hgy59 committed:
rename cross/expat to cross/libexpat to match upstream

hgy59 committed:
owntone: update dependencies

- update alsa-lib
- update libplist
- update libwebsockets
- update mxml

hgy59 committed:
revert update of cross/libplist

- libplist 2.2 is not compatible with libusbmuxd

hgy59 committed:
add initial service-setup.sh

hgy59 committed:
owntone: fix build for ARMv5 and PPC_ARCHS

hgy59 committed:
remove native/glib again

- native/glib is not required, since glib-mkenums is a python script and must not be built for the build system
- adapted from #5298

hgy59 committed:
owntone: fix error handling when avahi is not available

hgy59 committed:
revert unrelated changes

hgy59 committed:
owntone: adjust ffmpeg dependency folder (was renamed from ffmpeg to ffmpeg4)

hgy59 committed:
revert unrelated changes

hgy59 committed:
Merge branch 'master' of https://github.com/hgy59/spksrc into add_forked_daapd

hgy59 committed:
Merge branch 'master' of https://github.com/hgy59/spksrc into add_forked_daapd

hgy59 committed:
owntone: update v29.0 and depend on ffmpeg7

hgy59 committed:
owntone: fix dependency

hgy59 committed:
owntone: remove obsolete patch

hgy59 committed:
define unsupported archs

hgy59 committed:
Merge branch 'master' of https://github.com/hgy59/spksrc into add_forked_daapd

Commit:
Merge branch 'master' of https://github.com/hgy59/spksrc into add_forked_daapd

hgy59 committed:
owntone: update to v29.2

- update owntone
- use spk-meta.mk
